packages/client/src/views/glsp-svg-views.tsx

```diff
@@ -0,0 +1,109 @@
+        if (!this.isVisible(node, context)) {
+            return undefined;
+        }
+        if (context.targetKind === 'hidden') {
+            // When rendering the diamond the calculated bounds by the browser (BBox) are larger than what we actually want them to be
+            // To ensure that the node's bounds remains unchanged, we render a rectangle with the correct bounds during hidden rendering
+            return hiddenBoundingRect(node);
+        }
+        const diamond = new Diamond({ height: Math.max(node.size.height, 0), width: Math.max(node.size.width, 0), x: 0, y: 0 });
+        const points = `${point(diamond.topPoint)} ${point(diamond.rightPoint)} ${point(diamond.bottomPoint)} ${point(diamond.leftPoint)}`;
+        return (
+            <g>
+                <polygon
+                    class-sprotty-node={node instanceof GNode}
+                    class-sprotty-port={node instanceof GPort}
+                    class-mouseover={node.hoverFeedback}
+                    class-selected={node.selected}
+                    points={points}
+                />
+                {context.renderChildren(node)}
+            </g>
+        );
+    }
+}
+
+@injectable()
+export class GCircularNodeView extends ShapeView {
+    render(node: Readonly<GShapeElement & Hoverable & Selectable>, context: RenderingContext, args?: IViewArgs): VNode | undefined {
+        if (!this.isVisible(node, context)) {
+            return undefined;
+        }
+        if (context.targetKind === 'hidden') {
+            // When rendering the diamond the calculated bounds by the browser (BBox) are larger than what we actually want them to be
+            // To ensure that the node's bounds remains unchanged, we render a rectangle with the correct bounds during hidden rendering
+            return hiddenBoundingRect(node);
+        }
+        const radius = this.getRadius(node);
+        const center = Dimension.center(node.bounds);
+        return (
+            <g>
+                <circle
+                    class-sprotty-node={node instanceof GNode}
+                    class-sprotty-port={node instanceof GPort}
+                    class-mouseover={node.hoverFeedback}
+                    class-selected={node.selected}
+                    r={radius}
+                    cx={center.x}
+                    cy={center.y}
+                ></circle>
+                {context.renderChildren(node)}
+            </g>
+        );
+    }
+
+    protected getRadius(node: GShapeElement): number {
+        const d = Math.min(node.size.width, node.size.height);
+        return d > 0 ? d / 2 : 0;
+    }
+}
+
+export function hiddenBoundingRect(withBounds: BoundsAware): VNode {
+    // an element with attribute ATTR_BBOX_ELEMENT is used by the hidden bounds updater to determine the bounds
+    return <rect attrs={{ [ATTR_BBOX_ELEMENT]: true }} class-hidden={true} {...Dimension.fromBounds(withBounds.bounds)} />;
```

Comment:
Side note: Explicitly setting the hidden class should not be necessary. The bounding rect is only rendered in the hidden viewer anyways => its always hidden by default

Reply:
Good catch! It was from another approach I tried before, thank you for catching that!